### PR TITLE
[11561] Fixed Address- and RelationshipType for Zusatzaddress

### PR DESCRIPTION
**Pull request by ngiger**
The french translation introduced bugs. The relations are no longer correctly labelled.

* Added Unit-Tests
* Added the follown functions ch/elexis/core/types/LocalizeUtil.java
  *  ensureUseOAddressTypefMessages
  * ensureUseOfRelationShipMessages

This must be ported to branch 3.6 as well

### CODE REVIEW

File: bundles/ch.elexis.core.ui.contacts/src/ch/elexis/core/ui/contacts/views/Patientenblatt2.java [hunk 1]
```diff
@@ -790,7 +790,7 @@ public void refreshUi(){
 			+ StringTool.unNull(actPatient.getVorname()) + " (" //$NON-NLS-1$
 			+ actPatient.getPatCode() + ")"); //$NON-NLS-1$
 		inpAdresse.setText(actPatient.getPostAnschrift(false), false, false);
-		UserSettings.setExpandedState(ecZA, "Patientenblatt/Zusatzadressen"); //$NON-NLS-1$
```
Comment (col-panic) [inline bundles/ch.elexis.core.ui.contacts/src/ch/elexis/core/ui/contacts/views/Patientenblatt2.java]: Wieso das NON-NLS-1 raus? das zeit doch genau an, dass das nicht übersetzt werden soll

File: bundles/ch.elexis.core/src/ch/elexis/core/types/LocalizeUtil.java [hunk 2]
```diff
@@ -1,29 +1,69 @@
...
-import java.util.ResourceBundle;
-
-import org.eclipse.emf.common.util.Enumerator;
-
-import ch.elexis.core.interfaces.ILocalizedEnum;
-import ch.elexis.core.model.MaritalStatus;
-
-public class LocalizeUtil {
-	/**
-	 * Use with EMF generated not instances of {@link ILocalizedEnum}. Get the localized String for
```
Comment (col-panic) [inline bundles/ch.elexis.core/src/ch/elexis/core/types/LocalizeUtil.java]: I prefer implementing ILocalizedEnum for interfaces where a localization is relevant.

File: bundles/ch.elexis.core/src/ch/elexis/core/types/LocalizeUtil.java [hunk 3]
```diff
@@ -1,29 +1,69 @@
...
+
+import org.eclipse.emf.common.util.Enumerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ch.elexis.core.interfaces.ILocalizedEnum;
+import ch.elexis.core.l10n.Messages;
+
+public class LocalizeUtil {
+	 private static Logger log = LoggerFactory.getLogger(LocalizeUtil.class);
```
Comment (col-panic) [inline bundles/ch.elexis.core/src/ch/elexis/core/types/LocalizeUtil.java]: ?! please implement ILocalizedEnum for the resp. interfaces, not a separate utility class

**Comment by ngiger** [comment]
After discussing by phone we agreed that we should not merge my change for Refactored ZusatzAdresse/BezugsKontak to use ILocalizedEnum
as they are not Enums
Therefore pushed the other changes to the master and closing this pull request

**Pull request closed without merging**